aplib.net-demo/Assets/Scripts/Entities/Enemy/RangedEnemy.cs

```diff
@@ -0,0 +1,103 @@
+using Entities.Weapons;
+using System.Collections;
+using UnityEngine;
+
+/// <summary>
+/// Ranged enemy that attacks the player when in attack range.
+/// Has a separate vision range within which it will move closer to the player.
+/// </summary>
+[RequireComponent(typeof(Timer))]
+[RequireComponent(typeof(EntitySound))]
+public class RangedEnemy : DummyEnemy
+{
+    [SerializeField]
+    private float _attackCooldown = 2f;
+
+    [SerializeField]
+    private int _attackRange = 10;
+
+    [SerializeField]
+    private int _visionRange = 20;
+
+    private bool _movingCloser;
+    private Timer _attackTimer;
+    private RangedWeapon _rangedWeapon;
+    private EntitySound _entitySound;
```

Comment:
We do not want to store the EntitySound and use it in this class, requiring it to enforce it existing on an enemy is fine, but it should be used within the weapons as they were, so that they can be used more easily and do not have to be used for every enemy type we may introduce

(Right now it's easier, melee weapon = melee sound, ranged weapon = ranged sound, check this out on the main branch!)

this comment applies to all uses of entity sound in this script